feat: only allow lowercase slug configuration

quincydejong opened the pull request:
If identity-providers.json is configured with the slugs "EXAMPLE-1" and "example-1", it should throw an exception.
See feedback: https://github.com/privacybydesign/yivi-generic-identity-attestation-provider/pull/6#discussion_r2161546833

Also see: https://github.com/privacybydesign/yivi-generic-identity-attestation-provider/pull/6#discussion_r2161542596

However some parts of the app depend on it being case sensitive such as in IdentityProvidersExtension.cs.

So alternatively we could enforce all slugs to be lowercase and throw an exception otherwise.

EDIT: After discussion, enforce lowercase slugs only, but convert user input slugs to lowercase. So for example when a user navigates to '/EXAMPLE' it will be converted to '/example' and the user will still get their expected result.

quincydejong committed:
feat: check for slugs in a case-insensitive way

"example-1" and "EXAMPLE-1" should be counted as duplicated. See feedback: https://github.com/privacybydesign/yivi-generic-identity-attestation-provider/pull/6#discussion_r2161546833

quincydejong committed:
test: add case-insensitive duplicate slug test

Also update the original test with the new exception message.

kamphuisem commented:
Is the slug case sensitive in the URL or not?
Like, if you go to `<giap-url>/en/example-1` does it render the same as `<giap-url>/en/EXAMPLE-1`?

kamphuisem commented:
Shouldn't you also add lowercasing to the function in [GIAP.Server/Services/IdentityProviderService.cs](https://github.com/privacybydesign/yivi-generic-identity-attestation-provider/pull/6/files/2ccd2d9f5a3f3c24793052ea334ac4fbda77cb69#diff-74a405b2d28aebbda18d29ad1417e824b4f7c88e95ec7ba799e67d4f44cda6d8)?
If you go to `<giap-url>/en/EXAMPLE-1`, the slug will probably be uppercase in the controller, which might not find the IdP in the IdentityProviderService.

quincydejong committed:
refactor: only allow lowercase slugs

For user input convert the slugs to lowercase.
This is necessary because when a user navigates to /Example, it shouldn't respond that it didn't find anything.

kamphuisem commented on GIAP.Server/Services/IIdentityProviderService.cs:
```diff
@@ -17,9 +17,16 @@ public interface IIdentityProviderService
     /// <summary>
     /// Get an identity provider by its web url slug.
+    /// This method is used to find an IdentityProvider from a user input slug.
+    /// The user input slug is always converted to lowercase.
```
This is an implementation detail (in the class, not in the interface).
Remove this comment

quincydejong committed:
docs: remove implementation detail comment from summary docs
